HACKING.jst.md

```diff
@@ -0,0 +1,100 @@
+Jane Street `ocamlformat` instructions
+======================================
+
+This file describes the shape of the `ocamlformat` repo, with a
+specific eye toward making it easy to update `ocamlformat` to deal
+with Jane Street extensions.
+
+Overall structure
+-----------------
+
+The main implementation of `ocamlformat` lives in the `lib` directory.
+This is where the `ocamlformat` executable is driven.
+
+The most important file for our purposes here is `lib/Fmt_ast.ml`, which
+defines the functions that actually print the formatted AST.
```

Comment:
Some modules in this directory to mention, here or elsewhere:

- `Fmt` has combinators useful for printing.  You're likely to use this module but not super likely to change it.
- `Ast` has various helpers, mainly for (a) constructing contexts, and (b) analyzing whether terms need parentheses.  It's common to need to change the latter for new syntax.
- `Cmts` deals with comments (not the `cmt` file format)!  Comments are not placed in the extended AST, but rather maintained in a table that is checked at various points using the helpers in this module.
- `Sugar` has various bits of support code for sugaring/desugaring syntax (e.g., "multi-argument" functions).

I think those are the main ones I've had to modify or use.